src/js/ripple/amount.js

```diff
@@ -798,29 +576,41 @@ Amount.prototype.parse_quality = function(quality, counterCurrency, counterIssue
 
   var baseCurrency = Currency.from_json(opts.base_currency);
 
-  this._is_negative = false;
-  this._value       = new BigInteger(quality.substring(quality.length-14), 16);
-  this._offset      = parseInt(quality.substring(quality.length-16, quality.length-14), 16)-100;
+  var mantissa_hex  = quality.substring(quality.length-14);
+  var offset_hex    = quality.substring(quality.length-16, quality.length-14);
+  var mantissa      = new BigNumber(mantissa_hex, 16);
+  var offset        = parseInt(offset_hex, 16) - 100;
+
+  var value = new BigNumber(mantissa.toString() + 'e' + offset.toString());
+
   this._currency    = Currency.from_json(counterCurrency);
   this._issuer      = UInt160.from_json(counterIssuer);
   this._is_native   = this._currency.is_native();
 
+  var exponent = 0;
+  if (this._is_native) {
+    if (opts.inverse) {
+      exponent += 1;
+    } else {
+      exponent -= 1;
```

Comment:
Suggest either: 

```
if (opts.inverse) {
  value = value.toPower(-1);
}

// quality is `takerPays / takerGets`
// counterCurrency equates to takerPays
// We don't want to pay a million times too much
// because the quality is in terms of drops.
if (!opts.xrp_as_drops) {
  if (this._is_native) {
    value = value.div(Amount.bi_xns_unit);
  }
  else if ( baseCurrency.is_valid() && baseCurrency.is_native()) {
    value = value.times(Amount.bi_xns_unit);   
  }  
};

this._set_value(value);
```

or 

```  
var adjusted = opts.inverse ? value.toPower(-1) : value;

var native_scale = opts.xrp_as_drops ? adjusted :
                          // quality is `takerPays / takerGets`
                          // counterCurrency equates to takerPays
                          // We don't want to pay a million times too much
                           this._is_native ? adjusted.div(Amount.bi_xns_unit) :
                           baseCurrency.is_valid() && baseCurrency.is_native() ? 
                          // The quality is how much you must `pay` to get ONE of `gets`
                          // If takerGets was native then we need to pay a million 
                          // times more for each (one) unit. 
                           adjusted.times(Amount.bi_xns_unit) :
                           adjusted;

this._set_value(native_scale);
```